controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -579,17 +598,25 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 }
 
 // Ensure we have the information about the hardware on the host.
-func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
+func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner, info *reconcileInfo, force bool) actionResult {
+	info.log.Info("inspecting hardware")
 
 	if inspectionDisabled(info.host) {
 		info.log.Info("inspection disabled by annotation")
 		info.publishEvent("InspectionSkipped", "disabled by annotation")
 		return actionComplete{}
 	}
 
-	info.log.Info("inspecting hardware")
+	// Delete inspect annotation if exists
+	if hasInspectAnnotation(info.host) {
+		if clearInspectAnnotation(info.host) {
+			if err := r.Update(context.TODO(), info.host); err != nil {
+				return actionError{errors.Wrap(err, "failed to remove inspect annotation from host")}
+			}
+		}
+	}
 
-	provResult, details, err := prov.InspectHardware(info.host.Status.ErrorType == metal3v1alpha1.InspectionError)
+	provResult, details, err := prov.InspectHardware(force)
```

Comment:
Right, this is what was trying to avoid initially and why I used `forceNew`.

Reply:
But the current `force` flag is just based on annotation, so the triggering conditions are not the same of before. Looks like another relevant indication to keep the two things separated

Reply:
Yes sure. @zaneb do you think it is okay to create another flag, `forceNew` that can server re-inspection only? Because, relying on `force` would break the existing functionality as @andfasano mentioned [here](https://github.com/metal3-io/baremetal-operator/pull/607#discussion_r595418581)

Reply:
@fmuyassarov Very similar issue that I had in https://github.com/metal3-io/baremetal-operator/pull/817/commits/3e860bf5815ca7b0df41fd046fdc328d160034be (PR https://github.com/metal3-io/baremetal-operator/pull/817)
cc: @zaneb @andfasano

Reply:
You would at a minimum have to OR the two conditions together.